tests/common/dualtor/server_traffic_utils.py

```diff
@@ -0,0 +1,90 @@
+"""Utils to verify traffic between ToR and server."""
+import contextlib
+import logging
+import tempfile
+import sys
+import time
+
+from io import BytesIO
+from ptf.dataplane import match_exp_pkt
+from scapy.all import sniff
+from scapy.packet import ls
+
+
+@contextlib.contextmanager
+def dump_intf_packets(ansible_host, iface, pcap_save_path, dumped_packets, pcap_filter=None):
+    """Dump packets of the interface and save to a file."""
+    start_pcap = "tcpdump --immediate-mode -i %s -w %s" % (iface, pcap_save_path)
+    if pcap_filter:
+        start_pcap += (" " + pcap_filter)
+    start_pcap = "nohup %s  > /dev/null 2>&1 & echo $!" % start_pcap
+    pid = ansible_host.shell(start_pcap)["stdout"]
+    # sleep to let tcpdump starts to capture
+    time.sleep(1)
+    try:
+        yield
+    finally:
+        ansible_host.shell("kill -s 2 %s" % pid)
+        with tempfile.NamedTemporaryFile() as temp_pcap:
+            ansible_host.fetch(src=pcap_save_path, dest=temp_pcap.name, flat=True)
+            packets = sniff(offline=temp_pcap.name)
+            dumped_packets.extend(packets)
+
+
+class ServerTrafficMonitor(object):
+    """Monit traffic between ToR and server."""
```

Comment:
This tool should be able to capture traffic on any vmhost vlan interfaces that are connected to DUT. The DUT is not limited to ToR. It can be T1 and other roles.

Reply:
Yes, you're right, will rename it then.